### PR TITLE
Make varedited pumps/pipes keep their varedited part of their names

### PR DESCRIPTION
**Pull request by RandomGamer123**
## About The Pull Request
Atmos pumps/filters are showing up as just "omni gas pump" even if their name had been edited in the map editor beforehand. This PR fixes that by using the name at initialisation/construction rather than initial(name).

## Why It's Good For The Game
Fixes an undocumented bug, if a pump's name is varedited in the map it probably should actually show up.

## Changelog

:cl:
fix: Certain pumps/filters in atmospherics and engineering now show their more descriptive names which haven't been showing up properly for months.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
what actually ensures that atmos things retain their name here? it'll still just be overriden by update_name right?

**Comment by RandomGamer123** [comment]
Well this makes update_name only run once, the first time

**Comment by LemonInTheDark** [comment]
Oh I see, initial(name) -> name

**Comment by Toastgoats** [comment]
Holy shit thank you this was pissing me off for weeks

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 1]
```diff
@@ -461,7 +464,9 @@
 	SSair.add_to_rebuild_queue(src)
 
 /obj/machinery/atmospherics/update_name()
-	name = "[GLOB.pipe_color_name[pipe_color]] [initial(name)]"
+	if(!name_initialised)
```
Comment (ShizCalev) [inline code/modules/atmospherics/machinery/atmosmachinery.dm:467]: this will fuck up vents if they get moved from what i can tell

**Comment by ShizCalev** [comment]
there's a better way to handle this, lemme find the proc again

File: code/modules/atmospherics/machinery/atmosmachinery.dm (hunk 1 above)
Comment (ShizCalev) [inline code/modules/atmospherics/machinery/atmosmachinery.dm:467]:
```
var/custom_name //keeps the name of the object from being overriden during map load

/obj/machinery/atmospherics/Initialize(mapload)
	if(mapload && name != initial(name))
		custom_name = TRUE
	return ..()	
	
/obj/machinery/atmospherics/update_name()
	if(!custom_name)	
		name = "[GLOB.pipe_color_name[pipe_color]] [initial(name)]"
	return ..()
```

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 2]
```diff
@@ -461,7 +464,9 @@
...
+		name = "[GLOB.pipe_color_name[pipe_color]] [name]"
+		name_initialised = TRUE
```
Comment (ShizCalev) [inline code/modules/atmospherics/machinery/atmosmachinery.dm:469]: IIRC changing a var from default value does make the object use more memory, so doing this for EVERY pipe in the game on the off chance you _might_ have one that is vareditted is the reverse of how this should be handled.

Comment (ShizCalev) [reply code/modules/atmospherics/machinery/atmosmachinery.dm:469]: also this doesn't fix scrubbers / vents so you are aware.

File: code/modules/atmospherics/machinery/atmosmachinery.dm (hunk 1 above)
Comment (ShizCalev) [reply code/modules/atmospherics/machinery/atmosmachinery.dm:467]: oh right and  
```
/obj/machinery/atmospherics/vv_edit_var(vname, vval)
	if(vname == NAMEOF(src, name))
		custom_name = TRUE
	return ..()
```

File: code/modules/atmospherics/machinery/atmosmachinery.dm (hunk 2 above)
Comment (RandomGamer123) [reply code/modules/atmospherics/machinery/atmosmachinery.dm:469]: Yeah just looked at vents and scrubbers and those needs more changes

**Pull request merged**